### ISSUE TITLE
Suggestion: Swap piece colors on dark terminal background

### ISSUE THREAD

**Issue by zzril** [hrushikeshrv/chessengine#10]
When using a terminal with dark background, the colors of the pieces look a bit weird, i.e. the "black pieces" appear brighter than the "white pieces".

If possible, it would be nice to have a setting to swap the piece colors for terminals with dark background color. Or - even better - detect this automatically.

(Probably not top-priority atm, though. :D )

---

![Board image on a terminal with bright background.](https://abload.de/img/chessboard_bright_bac8kea8.png)

![Board image on a terminal with dark background. The "white pieces" on the bottom appear darker than the "black pieces" on the top.](https://abload.de/img/chessboard_colourst5ek4.png)

**Comment by Substancia** [comment]
Can I take this up (probably as my second Hacktoberfest PR since this isn't too priority)?

**Comment by hrushikeshrv** [comment]
I'm not too sure we can detect the background color on all terminals, can we? 
> Probably not top-priority atm, though  

I agree, I've asked @Montekkundan to update the format of the chessboard a little anyway, by also printing the ranks and files next to the board to make it a little easier to see which moves to make. Once that is merged, we can think about improving the contrast/appearance of the pieces

**Comment by Montekkundan** [comment]
@Substancia ill send a pr tomorrow for updating the chessboard format and the you can work on this.

**Comment by fr4nkln11** [comment]
I have solution for this problem, instead of trying to swap piece colour, can't we just let the player know which piece is which by printing a Unicode representation of the piece next to the name of the piece colour. So far we only know how this looks on black and white, but I think with this solution it wouldn't matter as long as the user knows which piece represents a colour.
I will create a PR for this now.

**Comment by hrushikeshrv** [comment]
IMO, I think this is a limitation of the terminal/Unicode, and I don't think there is much we can do in this situation. One idea is that instead of completely relying on the terminal, I think we could add an optional dependency to some GUI framework, and let users play a game using a GUI instead of on the terminal. But that's just an idea, for now I don't think we can do much about this issue.

**Comment by fr4nkln11** [comment]
What about my idea, and also before GUI, can we try Textualize/rich or some other terminal framework like click or argparse

**Comment by zzril** [comment]
I came across [this post on stackoverflow](https://stackoverflow.com/questions/2507337/how-to-determine-a-terminals-background-color) which indicates that it should be possible to detect the background color, at least on XTerm-compatible terminals.  
I have tested the control sequence on my own terminal, and it worked. :)

In order to integrate this into the chess program, we could use the shell script given in one of the answers to the Stackoverflow post and execute it suring startup of the chessengine. If it fails to execute, then we just default to the normal colors. Something like that.

Alternatively (or additionally), we could also use a config file, or simply let the user choose color schemes on the command-line. But having this auto-detection, at least in those cases where technically possible, would of course be the best (and most user-friendly) way to do it.

I can also work on that if noone else wants to. ;)

Edit: I guess it should also be possible to port the script to python and just do everything from within Python, so no need to spawn a subprocess or stuff like that.

**Comment by fr4nkln11** [comment]
Nice, that seems to be the best option "auto-detection"

**Comment by Montekkundan** [comment]
Nice work 😄 ! I guess this will work great. @hrushikeshrv what do you think?

**Comment by hrushikeshrv** [comment]
I'm a little doubtful if the automatic detection will work across different terminals and different platforms, but it would be good to have in the cases where it does work, so I guess we can start working on that. Then after I merge #26, we could add an option to the cli to swap piece colors.

@zzril should I assign this to you?

**Comment by zzril** [comment]
> @zzril should I assign this to you?

Yes, I'd be happy to work on this. :)
